src/admin.ts

```diff
@@ -0,0 +1,178 @@
+import { createWithdrawFromTreasuryInstruction } from '@metaplex-foundation/mpl-auction-house/dist/src/generated/instructions'
+import { Wallet } from '@metaplex/js'
+import { Connection, PublicKey, Transaction } from '@solana/web3.js'
+import { MarketplaceClient } from './marketplace'
+import { AuctionHouse, MarktplaceSettingsPayload } from './types'
+
+interface MarketplaceBaseParams {
+  logo: { uri: string; type?: string; name?: string }
+  banner: { uri: string; type?: string; name?: string }
+  subdomain: string
+  name: string
+  description: string
+  transactionFee: number
+  creators: { address: string }[]
+}
+
+export type EditMarketplaceParams = MarketplaceBaseParams & {
+  auctionHouses: { address: string }[]
+}
+
+export type EditTokensParams = MarketplaceBaseParams & {
+  originalAuctionHouses: AuctionHouse[]
+  tokens: { address: string }[]
+}
+
+export class AdminClient extends MarketplaceClient {
+  constructor(connection: Connection, wallet: Wallet) {
+    super(connection, wallet)
+  }
+
+  async editMarketplace({
```

Comment:
I'd like to house all the marketplace actions under the marketplace client.

marketplace.update(payload: MarketplaceSettingsPayload)

Lets start with a simpler action that requires the caller to pass in the entire marketplace payload. 

Its easier for the caller to patch an existing marketplace json than to reconstruct in every action.